src/compiler/checker.ts

```diff
@@ -15920,6 +15920,9 @@ namespace ts {
                             getTypeFromTypeNodeWorker(<TypeNode>node) === tp; // use worker because we're looking for === equality
                     case SyntaxKind.TypeQuery:
                         return true;
+                    case SyntaxKind.MethodDeclaration:
+                    case SyntaxKind.MethodSignature:
+                        return (!(node as FunctionLikeDeclaration).type && !!(node as FunctionLikeDeclaration).body) || !!forEachChild(node, containsReference);

```

Comment:
I think this is a little safer in the eventuality that `!!forEachChild(node, containsReference)` needs to become more complex, since it avoids that duplication.

```suggestion
                        if (!(node as FunctionLikeDeclaration).type && !!(node as FunctionLikeDeclaration).body)) {
                            return true;
                        }
                        // falls through
```

Reply:
Ick, fall through to the default? That's not great, if cases would need to get added after these in the future (and I can see things that would need additions here in the future - we could probably get better at interning other function types the same way I improved methods here).

Reply:
eh, maybe so. I just didn't like the way the long line + trailing `||` hid the tree walk.